docs/sdk/v3/guides/03-web3-development-basics.md

```diff
@@ -0,0 +1,157 @@
+designed to index data like that on the go and provide easy access to it.
+
+### TheGraph
+
+A well-known, standardized implementation of indexers is [TheGraph](https://thegraph.com/). It is used by several of the big protocols,
+including Uniswap, to index data and make it accessible to users and dashboards.
+
+Using TheGraph, you can either access open [subgraphs](https://thegraph.com/explorer) via the GraphQL querying language, or [create your own](https://thegraph.com/docs/en/developing/creating-a-subgraph/) and deploy it.
+
+Throughout the docs you will see how to interact with the Uniswap subgraph to fetch tick data and more without ever touching RPCs.
+
+An important note about Indexers though:
+
+While they can be helpful, you need to be aware that the ultimate source of truth comes from the RPCs. As reorgs happen and
+certain issues on indexers arise, there might be certain datapoints that are either not fully up-to-date or even
+completely wrong on Indexers. If you have a use-case that requires perfectly correct data all the time, use RPCs directly.
+If you are just doing data visualization or dashboards, use Indexers if they fit your use-case.
+
+Some popular subgraphs that you can try to fetch data from the blockchain easily (click on playground to give it a try):
+
+* [Uniswap Messari subgraph](https://thegraph.com/explorer/subgraphs/ELUcwgpm14LKPLrBRuVvPvNKHQ9HvwmtKgKSH6123cr7?view=Overview&chain=mainnet)
+* [Snapshot Subgraph](https://thegraph.com/explorer/subgraphs/3Q4vnuSqemXnSNHoiLD7wdBbGCXszUYnUbTz191kDMNn?view=Overview&chain=mainnet)
+* [Aave V3 Messari subgraph](https://thegraph.com/explorer/subgraphs/HB1Z2EAw4rtPRYVb2Nz8QGFLHCpym6ByBX6vbCViuE9F?view=Overview&chain=mainnet)
+
+## Smart Contract Development
+
+Smart contracts are typically developed using the [Solidity language](https://soliditylang.org/).
+
+There are VSCode plugins that make the development with Solidity easier. One of them is Juan Blanco's "Solidity" that you can find
+over [here](https://marketplace.visualstudio.com/items?itemName=JuanBlanco.solidity).
+
+You would typically also go for a developer suite like one of the following:
+
+* [Foundry](https://github.com/foundry-rs/foundry) - Fast and modern developer tools for smart contract engineers.
+* [Hardhat](https://github.com/NomicFoundation/hardhat) - Scriptable, easy and battle-tested JS developer suite for smart contracts.
+* (deprecated) Ganache - One of the original developer suites. Should be avoided as there are many issues with it.
+
+To read data from smart contracts, or interact with them, use the [client SDKs](#client-implementations) linked to above.
+They help you generate the necessary RPC calls to fetch data from / send transactions to the blockchain to interact with your
+smart contracts. You can read more about it in the [ethers.js docs](https://docs.ethers.org/v5/api/contract/contract/).
+
+## The Uniswap SDK
```

Comment:
There is no one "Uniswap SDK". This will lead people to again use the deprecated "uniswap-sdk" package and get confused. Instead call this "Uniswap JS SDKs" or something similar and list and explain multiple packages from the Uniswap github.
At the very least, explain quickly:
- v3-sdk
- sdk-core
- v2-sdk
- smart-router-sdk
- v3-core package
- v3-periphery package